interfaces/builtin/modem_manager.go

```diff
@@ -0,0 +1,1205 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
+)
+
+var modemManagerPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the ModemManager service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# To check present devices
+/run/udev/data/* r,
+/sys/bus/usb/devices** r,
+
+# Access to modem ports
+/dev/tty[^0-9]* rw,
```

Comment:
I was hoping that interfaces would solve this sort of thing and snapd would be more dynamic rather than hard-coding devices in this manner, but again, we aren't there yet. Without this dynamism, if there is another interface that needs /dev/tty* then two snaps using two different interfaces on the same system might cause problems for each other by accessing the same device. Can you add a FIXME for this? Also, what about /dev/ttyUSB*, is that handled differently?